python/cudf/cudf/_lib/binaryop.pyx

```diff
@@ -118,7 +118,7 @@ cdef binaryop_v_v(Column lhs, Column rhs,
 cdef binaryop_v_s(Column lhs, Scalar rhs,
                   binary_operator c_op, data_type c_dtype):
     cdef column_view c_lhs = lhs.view()
-    cdef scalar* c_rhs = rhs.c_value.get()
+    cdef scalar* c_rhs = rhs.get_uptr()._device_uptr.get()
```

Comment:
This pattern is repeated everywhere, would it make sense to introduce a `get_raw_ptr()` function instead?

Reply:
Had a long discussion with @shwina about this. In the end, we thought it was a little dangerous to return a raw pointer from this API. The reason is that from the perspective of whatever is working with this `raw_ptr` there is ambiguity about who owns the scalar being pointed to, and if it's even alive. Returning the unique_ptr and then forcing the caller to reach inside it and get the scalar itself makes the callers usage of this a little more deliberate.

Reply:
Does this work?
```
from cython.operator cimport dereference as deref
... some code ...
cdef scalar* c_rhs = deref(rhs.get_uptr())
```

Reply:
Good question let me check.

Reply:
I had a call with @cwharris and @shwina about this. Chris raised the point that the wrapper really doesn't provide any additional safety around the object ownership as one could easily just move the unique pointer out of the wrapper. As such what is really needed is a scalar `view` class that works similarly to column views. In lieu of that, we ended up thinking that a function that returns a raw pointer, as well as clear documentation on what *not* to do with the resulting raw pointer (freeing it etc), was the best solution for the time being.